### ISSUE TITLE
Randomly totally broken in Firefox

### ISSUE THREAD

**Issue by hyperknot** [frontend-collective/react-sortable-tree#332]
In Firefox, **sometimes**, react-sortable-tree can get into a totally locked up state. This means that:

1. Dragging doesn't work at all.
2. Visually it looks bad. Non dragged items are thought to be dragged, in the rowCancelPad state.
2. Some static nodes are thought to be 0.5 opacity nodes, like while dragging, but actually they are not dragged.
4. canDrop is actually not called. 

The worst part of this is that **reloading the page doesn't solve this issue**. Once a page gets into this state, it'll stay this forever. The only fix is to **open a new tab and close the old one**.

I'm trying to make a reproducible case, but my code is basically node-renderer-default and tree-node-default simplified. 

![sortable](https://user-images.githubusercontent.com/494223/40919394-ade57884-6809-11e8-8642-bf2ad43935e5.gif)

**Comment by hyperknot** [comment]
OK, I could reproduce it with the official examples on the official pages. 

One particular way to trigger it is to drag one node and then quickly double click. Playing around a bit with dragging and quickly double clicking can get into this state in Firefox.

It kind of thinks it's in a continuous dragging state.

![sortable](https://user-images.githubusercontent.com/494223/40920872-2810e20c-680e-11e8-9784-44e2eaf41b95.gif)

**Comment by hyperknot** [comment]
You can see how this broken state persist after page reloads.

![sortable](https://user-images.githubusercontent.com/494223/40921184-0c072ba6-680f-11e8-84a4-325e50649d4d.gif)

**Comment by wuweiweiwu** [comment]
This is super weird. I don't think its with the react component though. Initially I thought there was some state inconsistency. But the reload should reset the state. 

Thus I think the issue is with the HTML dnd backdend..

**Comment by hyperknot** [comment]
I was thinking along the way. Maybe opening an issue in dnd is a good idea.

**Comment by wuweiweiwu** [comment]
Definitely needs more investigation. I'm also thinking of getting rid of the dependency on react-dnd / html backend entirely.  possibly using whatever [react-sortable-hoc](https://github.com/clauderic/react-sortable-hoc) uses in the backend

**Comment by cormacrelf** [comment]
This is an awful bug in Firefox that is not fixed yet. https://github.com/react-dnd/react-dnd/issues/1000

You could try https://github.com/zyzo/react-dnd-mouse-backend -- it would skip over all the browser-specific HTML5 bugs that just keep popping up.

**Comment by wuweiweiwu** [comment]
Yeah... I wonder if its worth just writing everything manually. I'll talk to the author of `react-sortable-hoc`

I'll definitely check the mouse backend. Thanks @cormacrelf
